### PR TITLE
COMMERCE-11650

### PR DESCRIPTION
**Pull request by brianikim**
@andrea-ale-sbarra ,

https://issues.liferay.com/browse/COMMERCE-11650, tied to LPP.

We most likely need input from @alessiorendina  because he implemented https://issues.liferay.com/browse/COMMERCE-5205 .

I am not sure whether we should support this as a default behavior, but I went ahead and made an implementation so we can at least get a discussion going.

Sincerely,
Brian I. Kim

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-inventory-service/src/main/java/com/liferay/commerce/inventory/internal/method/DefaultCommerceInventoryMethodImpl.java [hunk 1]
```diff
@@ -150,10 +150,15 @@ public String getLabel(Locale locale) {
 	public int getStockQuantity(
 		long companyId, long commerceChannelGroupId, String sku) {
 
-		return Math.min(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-inventory-service/src/main/java/com/liferay/commerce/inventory/internal/method/DefaultCommerceInventoryMethodImpl.java:153]: this looks like a big change, IDK why it was made like this and thus IDK if this is the right fix.

**Review by andrea-ale-sbarra — COMMENTED** [review]
@alessiorendina @luca-pellizzon can you please have a look at this?

Thank you!

File: modules/apps/commerce/commerce-inventory-service/src/main/java/com/liferay/commerce/inventory/internal/method/DefaultCommerceInventoryMethodImpl.java (hunk 1 above)
Comment (luca-pellizzon) [reply modules/apps/commerce/commerce-inventory-service/src/main/java/com/liferay/commerce/inventory/internal/method/DefaultCommerceInventoryMethodImpl.java:153]: This is probably correct, but needs lots of testing

**Review by luca-pellizzon — APPROVED** [review]

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
Hi @brianikim could you please update the [modules-semantic-versioning-jdk8](https://testray.liferay.com/home/-/testray/case_results/2342605127?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2342602599%26testrayRunId%3D2342603316%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc), thanks!

cc/ @victoroscuro

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
Hi @brianikim, I created a new pr with the automated tests, we can see there if the stable job is still failing or not. Thank you!

**Pull request closed without merging**